src/server/command_registry.cc

```diff
@@ -129,6 +130,15 @@ CommandRegistry& CommandRegistry::operator<<(CommandId cmd) {
     k = is_sub_command ? absl::StrCat(it->second, " ", maybe_subcommand[1]) : it->second;
   }
 
+  if (is_sub_command) {
+    for (string name : GetFlag(FLAGS_restricted_commands)) {
+      auto upper_name = AsciiStrToUpper(name);
+      if (cmd.name().starts_with(upper_name)) {
+        restricted_cmds_.emplace(cmd.name());
```

Comment:
why are you editing the restricted_cmds_ instead of just calling SetRestricted(true) maybe_subcommand.front() is found in restricted_cmds_